### PR TITLE
Support subdomains configuration

### LINKED ISSUE #132: Implement multi domain support
We need to support multi domains on our backend. 

- Communities should be able to choose a subdomain (we need to have a blacklist filled from the migration)
- Backend should take the host into consideration when receiving requests

#### Context
https://github.com/cambiatus/improvement-proposals/issues/35

### PR DESCRIPTION
**Pull request by lucca65**
## What issue does this PR close
Closes #132 

## Changes Proposed ( a list of new changes introduced by this PR)
- [x] Adds new `domainAvailable` GraphQL query, to be used by the community form
- [x] Updates community schema to support subdomain
- [x] Adds new `subdomains` table, it stores used domains and work as a blacklist for the new `domainAvailable` query
- [x] Automatically adds existing communities entries to `subdomains` 
- [x] Automatically adds `staging` and `demo` urls to the blacklist
- [x] Add validation to domain values

## How to test ( a list of instructions on how to test this PR)
- `mix test` will do

### CODE REVIEW

**Review by venomnert — COMMENTED** [review]
I am getting some test errors. I will push a fix for these @lucca65

File: priv/repo/migrations/20210421235832_subdomain_initial_fill.exs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  def gen_subdomain(name) do
+    name
+    |> String.downcase()
+    |> :unicode.characters_to_nfd_binary()
+    |> String.replace(~r/\W/u, "")
+    |> String.split(" ")
+    |> hd
+    |> Kernel.<>(".cambiatus.io")
```
[the remark this answers is not in the document]
Comment (lucca65) [reply priv/repo/migrations/20210421235832_subdomain_initial_fill.exs:44]: I think so for the initial data, this way we can support more domains in the future

**Comment by lucca65** [comment]
Hey @venomnert I've waited during the day but since the fix was pretty quickly to do, I've worked on it!

**Comment by venomnert** [comment]
Sorry for the delay @lucca65. 

* Regarding the subdomain I am unclear as to how we are to associate a `subdomain` to the `community`?
* Also how we are ensuring a 1 - 1 relationship

**Comment by lucca65** [comment]
about the first question: it's a simple association to another table 

about the second: we don't! there will be more entries on subdomains than on community, to ensure we have it as a blacklist easily

**Comment by venomnert** [comment]
>about the first question: it's a simple association to another table

I understand that; however, on community creation I am not seeing any association being created on `Community.changeset`

**Comment by lucca65** [comment]
and you won't, community creation is not done using a changeset, but rather on event source

**Comment by venomnert** [comment]
Got it!

**Review by venomnert — APPROVED** [review]
This is good!

**Pull request merged**